easybuild/easyblocks/c/cuda.py

```diff
@@ -92,6 +92,16 @@ def __init__(self, *args, **kwargs):
         self.cfg.template_values['cudaarch'] = cudaarch
         self.cfg.generate_template_values()
 
+    def fetch_step(self, *args, **kwargs):
+        """Check for EULA acceptance prior to getting sources."""
+        # EULA for CUDA must be accepted via --accept-eula-for EasyBuild configuration option,
+        # or via 'accept_eula = True' in easyconfig file
+        self.check_accepted_eula(
```

Comment:
I'm fine with checking it here, but maybe we should also check again in `install_step` just in case the install file is already available?

I would do that via a custom method `check_accepted_cuda_eula` in this easyblock, which we call twice (in `fetch_step` + `install_step`, or maybe `extract_step` rather than `install_step`)

Reply:
Actually, nevermind, `fetch_step` is always run, even when the install files is already available, as @branfosj pointed out to me, so should be good as is.